src/User/Filter/AccessRuleFilter.php

```diff
@@ -13,15 +13,42 @@
 
 use Closure;
 use Da\User\Model\User;
+use Da\User\Traits\ModuleAwareTrait;
 use yii\filters\AccessRule;
 
 class AccessRuleFilter extends AccessRule
 {
+    use ModuleAwareTrait;
+
+    public function allows($action, $user, $request)
+    {
+        $consentAction = 'user/settings/consent';
+        if (!$user->isGuest && $action->uniqueId !== $consentAction) {
+            $module = $this->getModule();
+
+            if ($module->GDPRrequireConsentToAll) {
+                $excludedUrls = $module->GDPRconsentExcludedUrls;
+                $excludedUrls[] = $module->GDPRprivacyPolicyUrl;
+                foreach ($excludedUrls as $url) {
+                    if (!fnmatch($url, $action->uniqueId)) {
+                        $identity = $user->identity;
+                        if (!$identity->gdpr_consent) {
+                            \Yii::$app->response->redirect([$consentAction]);
```

Comment:
`return` or `->send()` is [required](https://www.yiiframework.com/doc/api/2.0/yii-web-response#redirect()-detail)